src/App.tsx

```diff
@@ -5,10 +5,24 @@ import Header from './Header';
 import Login from './Login';
 import React, { SetStateAction, useEffect, useState } from 'react';
 import { supabase } from './setupSupabase';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
 
 function App() {
   const { mode, renderHeader } = Header();
   const [session, setSession] = useState(null);
+  const [userId, setUserId] = useState<string>('');
+  const [matches, setMatches] = useState<Array<any>>([]);
+
+  const getMatch = async () => {
+    try {
+      const { data, error } = await supabase.rpc('f_getmatchinfo', {
```

Comment:
I think the creation of a postgres function to do a join is a bit overkill, what benefits do you see doing so ?